### ISSUE TITLE
Evaluate moving from docker to podman

### ISSUE THREAD

**Issue by mairin** [FNNDSC/ChRIS_ultron_backEnd#227]
The setup for the ChRIS Backend and the ChRIS store backends involves using docker and docker-compose. Because a number of our volunteers were running newer versions of Fedora, we ran into issues getting ChRIS running locally. Details of that are here:

https://github.com/FNNDSC/ChRIS_ultron_backEnd/wiki/Fedora-32-Support-for-Docker

This ticket is to evaluate what would be involved in moving this setup from docker/docker-compose to a pod-based podman setup. See https://www.redhat.com/sysadmin/compose-podman-pods for an example of what you'd be looking to evaluate us moving to.

**Comment by mairin** [comment]
(I think we're using this as a tracker ticket so I am going to add anyone working on subtickets as an assignee - don't panic! :) )

**Comment by junaruga** [comment]
@bradscalio thanks for your feedback.

#213 was already merged. So, it is not related to this ticket, right?
If #227 (this ticket) and #228 are interdependent, we can think which ticket is the more prioritized single task, and what part of the tasks we can work in parallel.

> the desired user story/experience for developers as well as the range of target deployments/needs for the ChRIS Backend and ChRIS store backends.

For me, the user experiences are 

* It works on Ubuntu/Mac OSX we are currently supporting 
* It works on Fedora as a default OS settings are that cgroups v2, and SE Linux enabled (mode: enforcing).

**Comment by junaruga** [comment]
> If #227 (this ticket) and #228 are interdependent, we can think which ticket is the more prioritized single task that we focus at first, and what part of the tasks we can work in parallel.

I think it's better to focus to work on the single task: #228 (evaluate moving from docker-compose to k8s or mini-kube on docker) at first, then work on #227 (from docker to podman).

Because when we try to work on #227, we need to verify podman-compose for podman. But according to [the research](https://github.com/FNNDSC/ChRIS_ultron_backEnd/wiki/Fedora-32-Support-for-Docker), podman-compose does not work for this repository's `docker-compose.yml`.

**Comment by mairin** [comment]
@junaruga podman-compose i think is a WIP without full functionality. what dwalsh recommended was a pod-based setup to replace the docker-compose command. This outlines the process pretty well i think if i understand correctly: https://www.redhat.com/sysadmin/compose-podman-pods <= this actually came out after we started talking about this!

**Comment by junaruga** [comment]
Sure. So, it's up to Brad and Pratik about how to proceed comfortably. I am okay for you to close #228 as a duplicated ticket, if you like it.

**Comment by bradscalio** [comment]
Thanks for the clarification.

Will start under the assumptions:

- Ensure the current developer experience and workflow when running on Ubuntu and Mac OS X translates for a developer wanting to use Red Hat Enterprise Linux 8+, Fedora 32+ or other similar RPM-based distributions.  
- The current developer experience when attempting on Fedora 32 is detailed at [Fedora 32 Support for Docker.](https://github.com/FNNDSC/ChRIS_ultron_backEnd/wiki/Fedora-32-Support-for-Docker)
- Replace technology where relevant with similar functionality implemented within the distribution.
- Follow secure by default and identify issues as they arise, provide workaround and separate issue if LOE for identified issue is large.

**Comment by mairin** [comment]
@bradscalio I just want to add that i ran everything described here https://fedoramagazine.org/docker-and-fedora-32/ on F31 and after that had no issues w running anything ChRIS. I am on F32 now and will test but I have heard from others it does work as well. Jennings wrote up vagrant instructions to work around it but at least on my system vagrant is a bit too heavyweight and slow so i (certainly biasedly) dont think a good long term solution.
